config.go

```diff
@@ -137,7 +137,7 @@ func (c *Config) Interface(num, alt int) (*Interface, error) {
 	}
 
 	// Select an alternate setting if needed (device has multiple alternate settings).
-	if len(c.Desc.Interfaces[num].AltSettings) > 1 {
+	if len(c.Desc.Interfaces[num-1].AltSettings) > 1 {
```

Comment:
Surely the whole if statement should loop over all of the `c.Desc.Interfaces` or just [0] zero element it ... if it exists ... e.g., `c.Desc.Interfaces[0]`.

Reply:
I think the solution proposed in the PR is obviously wrong -- if num == 0, this expression is equivalent to [-1], which will always fail.

> Surely the whole if statement should loop over all of the c.Desc.Interfaces or just [0] zero element it ... if it exists ... e.g., c.Desc.Interfaces[0].

[0] would also be obviously wrong, since it is not necessarily the same interface as was requested through `num`.

If my guess at the underlying root cause is correct - your device likely has the interface numbers that are not aligned with slice indices (see https://github.com/google/gousb/issues/65). The non-default alternate setting code path was missed when we were fixing the original issue.

The proper solution is indeed to loop over all c.Desc.Interfaces and find an interface with a matching number. I think it warrants splitting the `intfDesc()` helper (line 52) into two: one - `intfDesc(num int)` - to find the matching interface number, returning a `*Interface`; another one - `altSetting(intf *Interface, alt int)`, returning a `*InterfaceSetting`.

Reply:
@zagrodzki The configurationNumber (where num comes from) and the Interfaces are two COMPLETELY separate things. It should either be zero indexed or looped over. I will amend the MR to loop, but what, if any, value does that add? We should let the user specify an interfaceNumber imho, defaulted to zero.

Reply:
But at this stage, to avoid introducing a breaking change, it would either have to be specified with interfaceNumber or defaulted to configurationNumber.

Reply:
I'm not sure what this comment means. This is a part of the "Interface" method, https://pkg.go.dev/github.com/google/gousb#Config.Interface:

"num specifies the number of an interface to claim, and alt specifies the alternate setting number for that interface."

The `num` is the interface number (i.e. the `bInterfaceNumber` of the interface to claim), not a config number. `alt` is the `bAlternateSetting` of the requested interface descriptor.

Reply:
@zagrodzki I think you're confusing the interface number, e.g., 1, which is a descriptor, with selecting an interface. The loop is the correct way forward.

Reply:
@zagrodzki I have amended my original commit to loop over interfaces, I can confirm that this is working:

![Screenshot 2024-07-17 at 16 25 54](https://github.com/user-attachments/assets/2aacb5ab-992c-47c9-b3fd-e49d43b5b4b3)

What other verification would you like?